### PR TITLE
Add a filter to remove mutation

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #52614 Avoid dispatch overhead in call to mkldnn convolution
* #52613 Include max pool in fusion groups
* #52513 Extend subgraph utils to cover merging a node following a subgraph
* #52512 Remove unused subgraph vmap api
* #51924 Only functionalize ops which we want to include in mkldnn group
* **#51923 Add a filter to remove mutation**
* #51922 Dont error on 0-dim in convolution
* #51921 Add 0-dim support for binary mkldnn ops
* #51736 Handle mkldnn broadcasting in mkldnn fuser
* #51600 Add MKLDNN fuser
* #52786 [NNC] Make NNC sanitize input names
* #51484 Add support for linear in mkldnn fusion
* #51483 Move conv to mkldnn

Differential Revision: [D26696700](https://our.internmc.facebook.com/intern/diff/D26696700)

### CODE REVIEW

File: test/cpp/jit/test_misc.cpp [hunk 1]
```diff
@@ -2376,5 +2377,25 @@ TEST(ComputeFlopsTest, Basic) {
...
+  auto graph = std::make_shared<Graph>();
+  std::unordered_map<std::string, Value*> vmap;
+  parseIR(
+      R"IR(
+graph(%x.1 : Tensor):
+  %2 : int = prim::Constant[value=1]() # /data/users/eellison/pytorch/test/jit/test_remove_mutation.py:76:20
+  %9 : int = prim::Constant[value=4]() # /data/users/eellison/pytorch/test/jit/test_remove_mutation.py:79:20
+  %x.3 : Tensor = aten::add(%x.1, %2, %2) # /data/users/eellison/pytorch/test/jit/test_remove_mutation.py:76:16
+  %7 : Tensor = aten::add_(%x.3, %2, %2) # /data/users/eellison/pytorch/test/jit/test_remove_mutation.py:77:12
+  %y.1 : Tensor = aten::add(%x.3, %9, %2) # /data/users/eellison/pytorch/test/jit/test_remove_mutation.py:79:16
```
Comment (ZolotukhinM) [inline test/cpp/jit/test_misc.cpp:2390]: Nit: remove the comments to make it easier to read :)

**Review by ZolotukhinM — APPROVED** [review]
Looks good to me!

**Review by Krovatkin — APPROVED** [review]
:shipit:

**Review by Krovatkin — APPROVED** [review]
:shipit:

**Pull request closed without merging**